master/buildbot/test/unit/test_db_buildsets.py

```diff
@@ -175,7 +175,8 @@ def test_getBuildset_incomplete_zero(self):
                                       reason='rsn', sourcestamps=[234],
                                       submitted_at=datetime.datetime(1978, 6, 15, 12, 31, 15,
                                                                      tzinfo=UTC),
-                                      complete=False, complete_at=None, results=-1,
+                                      complete=False,
+                                      complete_at=epoch2datetime(0), results=-1,
```

Comment:
why not None?

Reply:
Because 0 comes from the fake reactor - it's still a correct date/time, the only thing special with it is that it's the default start time for fake reactor. `None` would mean that the date is not set altogether which is not the case.

Reply:
yes, but when complete=False, the complete_at should be None, not 0.
So I guess this is hidding something not good.
We have to differentiate null values (unset) to 0 value (start of times)

Reply:
Indeed, I completely missed that point. Agreed, needs investigation.

Reply:
@tardyp: Alright, so what's happening here is that we intentionally push `complete=0, complete_at=0` to the DB and check what `self.db.buildsets.getBuildset()` returns. Do we want to perform cleanup at the DB layer to return data that makes sense even if the data in the DB doesn't? As far as I see there's no other cases of such cleanup being performed.

Reply:
This test verifies that the data api converts zero value to null value. So as we are removing this feature, I would remove the test case.

Reply:
Fixed in 543ad404de5.